### PR TITLE
Add padding image transformation

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Adds padding to the image transforms library. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by NielsRogge — APPROVED** [review]
LGTM!

So, when having [the following code](https://github.com/mv-lab/swin2sr/blob/7eeebfba849bbc934ea254ec4cfa8e9d6fc0672c/models/network_swin2sr.py#L887-L892):

```
import torch.nn.functional as F

def check_image_size(self, x):
   _, _, h, w = x.size()
   mod_pad_h = (self.window_size - h % self.window_size) % self.window_size
   mod_pad_w = (self.window_size - w % self.window_size) % self.window_size
   x = F.pad(x, (0, mod_pad_w, 0, mod_pad_h), 'reflect')
   return x
```
Is this equivalent to the following?
```
from transformers.image_transforms import pad

mod_pad_h = (self.window_size - h % self.window_size) % self.window_size
mod_pad_w = (self.window_size - w % self.window_size) % self.window_size

x = pad(x, padding=[(0, 0), (0, 0), (0, mod_pad_w), (0, mod_pad_h)], mode='reflect')
```

Wonder whether we can add some tests to check the equivalence between PyTorch and NumPy implementations of padding?

Update: verified the equivalence in a [notebook](https://colab.research.google.com/drive/1fNl31IlzEwQLjICW7Vs3YDM4sMk-gxZG?usp=sharing).

**Comment by amyeroberts** [comment]
@NielsRogge I've asked for re-review as the logic has changed a bit from when you first reviewed.

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

File: src/transformers/image_transforms.py [hunk 1]
```diff
@@ -594,17 +588,19 @@ def pad(
...
         image (`np.ndarray`):
             The image to pad.
         padding (`int` or `Tuple[int, int]` or `Iterable[Tuple[int, int]]`):
-            Padding to apply to the edges of each axis. ((before_1, after_1), … (before_N, after_N)) unique pad widths
-            for each axis. ((before, after),) yields same before and after pad for each axis. (pad,) or int is a
-            shortcut for before = after = pad width for all axes.
+            Padding to apply to the edges of the height, width axes. Can be one of three formats:
+            - ((before_height, after_height), (before_width, after_width)) unique pad widths for each axis.
+            - ((before, after),) yields same before and after pad for height and width.
+            - (pad,) or int is a shortcut for before = after = pad width for all axes.
```
[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/image_transforms.py]: Yep, sorry I forgot to format these. Updated now.

**Review by NielsRogge — APPROVED** [review]
Thanks for adding! Do you think it'd be useful to add a pytorch padding vs our implementation equivalence test?

**Comment by amyeroberts** [comment]
> Thanks for adding! Do you think it'd be useful to add a pytorch padding vs our implementation equivalence test?

@NielsRogge What do you think should be covered wrt equivalence - is there logic you want to make sure is always aligned between the two? These transformations aren't meant to be a np copy of the torch library so there isn't a 1:1 mapping.

**Comment by amyeroberts** [comment]
@NielsRogge I'm going to merge. If we decide to add the equivalence tests I'll add in a follow up PR.

**Comment by NielsRogge** [comment]
Ok fine, it's just that I was working on a model (Swin2SR) that leverages torch.nn.functional.pad as seen [here](https://github.com/mv-lab/swin2sr/blob/7eeebfba849bbc934ea254ec4cfa8e9d6fc0672c/models/network_swin2sr.py#L891). So it'd be nice to check equivalence.
